modules/darwin/hm.nix

```diff
@@ -0,0 +1,53 @@
+{
+  mkTarget,
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+mkTarget {
+  name = "darwin";
+  humanName = "Darwin";
+  extraOptions =
+    {
+      image,
+    }:
+    {
+      home.activation = {
+        stylixBackground = ''
+          run ${pkgs.callPackage ./desktoppr.nix { }}/bin/desktoppr all ${image}
+
+        '';
+      };
+    };
```

Comment:
> > i'm getting some errors to do with callPackage not existing
> 
> this is probably due to the doc eval, cc @MattSturgeon

`extraOptions` is for declaring options. This looks a lot more like a config definition (you haven't used `lib.mkOption` or any of its wrappers).

The reason `pkgs.callPackage` doesn't exist when declaring options is to ensure it is not accidentally used in option docs (i.e. option examples, defaultText, etc). See #1212

Reply:
i'm familiar with the distinction and have written many modules, i was just very unsure how this mkStylixModule system works